extensions/amp-story/1.0/amp-story-parallax-service.js

```diff
@@ -0,0 +1,427 @@
+    });
+  }
+
+  /**
+   * Calculate the smoothed change in position using the device orientation
+   * change event then update each of the layers will its calculated position.
+   * @param {Event} event
+   * @private
+   */
+  parallaxOrientationMutate_(event) {
+    const window = this.win_;
+    const {screen} = window;
+
+    const page = this.parallaxPages_.filter(page => page.shouldUpdate())[0];
+    if (!page) {
+      return;
+    }
+
+    let {gamma, beta} = event;
+
+    // Detect the implementation of orientation angle
+    const angle =
+      'orientation' in screen ? screen.orientation.angle : screen.orientation;
+
+    // Reverse gamma/beta if the device is in landscape
+    if (window.orientation == 90 || window.orientation == -90) {
+      const tmp = gamma;
+      gamma = beta;
+      beta = tmp;
+    }
+
+    // Flip signs of the angles if the phone is in 'reverse landscape' or
+    // 'reverse portrait'
+    if (angle < 0) {
+      gamma = -gamma;
+      beta = -beta;
+    }
+
+    // Smooth the gamma value (X-AXIS)
+    if (this.smoothingPointsX_.length > SMOOTHING_PTS) {
+      this.smoothingPointsX_.shift();
+    }
+    this.smoothingPointsX_.push(gamma);
+    const avgX = sum(this.smoothingPointsX_) / SMOOTHING_PTS;
+    if (this.smoothingPointsX_.length > SMOOTHING_PTS && this.middleX_ == 0) {
+      this.middleX_ = avgX;
+    }
+
+    // Smooth the beta value (Y-AXIS)
+    if (this.smoothingPointsY_.length > SMOOTHING_PTS) {
+      this.smoothingPointsY_.shift();
+    }
+    this.smoothingPointsY_.push(beta);
+    const avgY = sum(this.smoothingPointsY_) / SMOOTHING_PTS;
+    if (this.smoothingPointsY_.length > SMOOTHING_PTS && this.middleY_ == 0) {
+      this.middleY_ = avgY;
+    }
+
+    let mappedX = avgX - this.middleX_;
+    let mappedY = avgY - this.middleY_;
+
+    // Limit the range for a smoother/less shaky effect
+    mappedX = mapRange(mappedX, -75, 75, -25, 25);
+    mappedY = mapRange(mappedY, -75, 75, -25, 25);
+
+    this.parallaxPages_.forEach(page => {
+      if (page.shouldUpdate()) {
+        if (this.middleY_ != 0 && this.middleX_ != 0) {
+          page.update(mappedX, mappedY);
+        } else {
+          page.update(this.middleX_, this.middleY_);
+        }
+      } else {
+        page.resetStyles();
+      }
+    });
+  }
+
+  /**
+   * Calculate the smoothed change in position using the mouse's position
+   * then update each of the layers will its calculated position.
+   * @param {Event} event
+   * @private
+   */
+  parallaxMouseMutate_(event) {
+    const page = this.parallaxPages_.filter(page => page.shouldUpdate())[0];
+    if (!page) {
+      return;
+    }
+
+    const percentageX =
+      (event.pageX -
+        (page.element./*REVIEW*/ offsetLeft -
+          page.element./*REVIEW*/ offsetWidth / 2)) /
+      page.element./*REVIEW*/ offsetWidth;
+    const percentageY =
+      (event.pageY - page.element./*REVIEW*/ offsetTop) /
+      page.element./*REVIEW*/ offsetHeight;
```

Comment:
Please replace all the `offsetTop` and `offsetHeight` by the results from `page.getLayoutBox()`, which are the cached measurements from the AMP runtime.